### PR TITLE
Define the called workflow using an expression

### LINKED ISSUE #3177: I can define the called workflow using an expression
**Is your feature request related to a problem? Please describe.**
I want to define the called workflow of a call activity on runtime using an expression.

**Describe the solution you'd like**
* I can define the BPMN process id of the called workflow using an expression (e.g. refer a variable)
* a call activity must have either a static process id or an expression
* the expression is evaluated when the call activity is entered

**Describe alternatives you've considered**
* combine static process id and expression into one attribute (like in Camunda 7) -> more confusing for the simple and more common static option 

**Additional context**
The BPMN XML could look like:

```xml
<bpmn:callActivity id="Task_09pdy54">
    <bpmn:extensionElements>
        <zeebe:calledElement processIdExpression="myProcessVariable" />
     </bpmn:extensionElements>
</bpmn:callActivity>
```

requires #3174

### PR DESCRIPTION
**Pull request by saig0**
## Description

* add processIdExpression attribute to model API
* deploy a workflow with call activity and process id expression
* the expression is evaluated when the call activity gets activated
* create incident if variable doesn't exist or is not a string
* extract common path expression validator

## Related issues

closes #3177 

## Pull Request Checklist

- [x] All commit messages match our [commit message guidelines](https://github.com/zeebe-io/zeebe/blob/develop/CONTRIBUTING.md#commit-message-guidelines)
- [x] The submitting code follows our [code style](https://github.com/zeebe-io/zeebe/wiki/Code-Style)
- [x] If submitting code, please run `mvn clean install -DskipTests` locally before committing

### CODE REVIEW

File: bpmn-model/src/main/java/io/zeebe/model/bpmn/validation/zeebe/ZeebeCalledElementValidator.java [hunk 1]
```diff
@@ -30,13 +37,21 @@
...
-      validationResultCollector.addError(
-          0,
-          String.format(
-              "Attribute '%s' must be present and not empty", ZeebeConstants.ATTRIBUTE_PROCESS_ID));
+    final Optional<String> processId =
+        Optional.ofNullable(element.getProcessId()).filter(notEmpty());
+    final Optional<String> processIdExpression =
+        Optional.ofNullable(element.getProcessIdExpression()).filter(notEmpty());
+
+    if (processId.isPresent() && processIdExpression.isPresent()) {
```
Comment (MiguelPires) [inline bpmn-model/src/main/java/io/zeebe/model/bpmn/validation/zeebe/ZeebeCalledElementValidator.java]: The two cases of this if are equivalent to:

```suggestion
    if (processId.isPresent() == processIdExpression.isPresent()) {
```

File: bpmn-model/src/main/java/io/zeebe/model/bpmn/validation/zeebe/ZeebeCalledElementValidator.java [hunk 2]
```diff
@@ -30,13 +37,21 @@
...
       final ZeebeCalledElement element, final ValidationResultCollector validationResultCollector) {
-    final String processId = element.getProcessId();
 
-    if (processId == null || element.getProcessId().isEmpty()) {
-      validationResultCollector.addError(
-          0,
-          String.format(
-              "Attribute '%s' must be present and not empty", ZeebeConstants.ATTRIBUTE_PROCESS_ID));
+    final Optional<String> processId =
+        Optional.ofNullable(element.getProcessId()).filter(notEmpty());
```
Comment (MiguelPires) [inline bpmn-model/src/main/java/io/zeebe/model/bpmn/validation/zeebe/ZeebeCalledElementValidator.java]:
```suggestion
        Optional.ofNullable(element.getProcessId()).filter(Predicate.not(String::isEmpty));
```

This way we can remove the `notEmpty` method

File: bpmn-model/src/main/java/io/zeebe/model/bpmn/validation/zeebe/ZeebeCalledElementValidator.java [hunk 3]
```diff
@@ -30,13 +37,21 @@
...
+    final Optional<String> processIdExpression =
+        Optional.ofNullable(element.getProcessIdExpression()).filter(notEmpty());
```
Comment (MiguelPires) [inline bpmn-model/src/main/java/io/zeebe/model/bpmn/validation/zeebe/ZeebeCalledElementValidator.java]:
```suggestion
        Optional.ofNullable(element.getProcessIdExpression()).filter(Predicate.not(String::isEmpty));
```

**Review by miguelpires — CHANGES_REQUESTED** [review]
Looks good. Just pointed out a couple simplifications that could be made

File: bpmn-model/src/main/java/io/zeebe/model/bpmn/validation/zeebe/ZeebeCalledElementValidator.java (hunk 2 above)
Comment (saig0) [reply bpmn-model/src/main/java/io/zeebe/model/bpmn/validation/zeebe/ZeebeCalledElementValidator.java]: Unfortunately, `Predicate.not()` is not available in Java 8 :sweat_smile: I was added with Java 11. So, we can't use it in the model API.

**Comment by saig0** [comment]
@MiguelPires, thank you for the feedback. I addressed your hints. Please have a look :)

File: bpmn-model/src/main/java/io/zeebe/model/bpmn/validation/zeebe/ZeebeCalledElementValidator.java (hunk 2 above)
Comment (MiguelPires) [reply bpmn-model/src/main/java/io/zeebe/model/bpmn/validation/zeebe/ZeebeCalledElementValidator.java]: Ah I see, I thought we were only using Java 8 in the client

**Review by miguelpires — APPROVED** [review]

File: bpmn-model/src/main/java/io/zeebe/model/bpmn/validation/zeebe/ZeebeCalledElementValidator.java (hunk 2 above)
Comment (saig0) [reply bpmn-model/src/main/java/io/zeebe/model/bpmn/validation/zeebe/ZeebeCalledElementValidator.java]: Correct but the model API is also used by the client :sweat_smile:

**Pull request merged**